### DISCUSSION TITLE
How do I deal with the a negative subdevice?

### DISCUSSION THREAD

**Discussion by 88fingerslukee** [probonopd/irdb#154] [Q&A] [answered]
Hi all, I'm trying to convert some IRDB codes to pronto codes using IRPTransmogrifier (from the command line) and I'm getting the following error:

`Failed to generate Pronto code, error: Parameter S = -1 is outside of the allowed domain: 0..255`

Any idea on how I'm supposed to render a code that has a negative subdevice?

**Comment by bengtmartensson** [comment, marked as answer]
-1 simply means "this parameter is not present". So in IrpTransmogrifier, just leave out S.

Another name (like null in Java, NULL in C, null_ptr in C++, None in Python, nil in Pascal,...) might have been less prone to misunderstandings...

**Reply by probonopd** [reply]
Thanks @bengtmartensson.

**Reply by 88fingerslukee** [reply]
Thanks to both of you, this worked!
